Closes: #244, #233, #262, #318

manit-alokbatham opened the pull request:
Closes #244, #233, #262, #318

<!-- 
Thank you for contributing to ODK!
-->

#### What has been done to verify that this works as intended?
Have a look: fixed along with new updates, and handled Security Exceptions, and other warnings.
https://drive.google.com/file/d/1TGT0vvEMRE7h-_ssFlpjdqnOZ2zDnFWs/view?usp=sharing

#### Why is this the best possible solution? Were any other approaches considered?
Using TextWatcher is a good approach if we need to check text changes in a particular EditText. There are some other solutions using Observer. And LiveData. As i've seen the current architecture of application. This approach is meaningful with less changes.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
I considered the same approach of activation Hotspot in Samsung M20 device, and given almost same behavior while making changes in current code. This looks better now with extra UE.

#### Before submitting this PR, please make sure you have:
- [Yes ] run `./gradlew checkCode` and confirmed all checks still pass OR confirm CircleCI build passes
- [ Yes] verified that any code or assets from external sources are properly credited in comments and/or in the [about file]
![device-2019-11-28-194507](https://user-images.githubusercontent.com/58147504/69813164-aaf5ba80-1217-11ea-8dcf-33a263567e8f.png)
(https://github.com/opendatakit/share/blob/master/share_app/src/main/assets/open_source_licenses.html).

--------First look when dialog becomes visible on the screen------------

lakshyagupta21 commented:
@manit-alokbatham Sorry but for the issue you raised the PR is already raised and the issue is claimed by @devanshi7799,  so I've to close the PR.  And I just wanted to give general suggestions on the PR that please try to gove some meaningful title to the PRs.  Issues number and PRs number doesn't help much.  And don't mention  PR numbers against `closes` in PR description.  And you can avoid uploading things on drive,  you can directly add them on the github issues and PRs.

manit-alokbatham commented:
@lakshyagupta21  

I'm new in here. Tried to set limit for password and added these changes to Hotspot password setting. This is first time i'm working in open source project, found interesting that's why made changes. Along with these changes, I made some changes to fix build failure issue. You can check that even the opendatakit/master's source is failing at build ./gradlew checkCode. I'm new here and I don't have any idea about pull request and all media upload. In future i'll take care of this. 

I had a talk on slack with @devanshi7799 yesterday, and se told me that codeCheck is failing. I've given efforts and fixed these failures too along with this.

lakshyagupta21 commented:
I've also gone through this phase during my early days of open source and we learn by making mistakes so its good to make mistakes but we should also  learn from them,  I can completely understand your interest in the Password issue and its good to see that you both are trying to resolve the issue but I would encourage both you and @devanshi7799 to discuss this on skunkworks channel on slack so that we all can discuss the issue and provide our inputs.
